spec/decorators/address_form_decorator_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'rails_helper'
+
+RSpec.describe AddressFormDecorator do
+  subject { described_class.new(form_object) }
+
+  let(:form_object) { double('FormObject', record: address_record) }
+
+  let(:address_record) { instance_double(Address, type: address_type, person: person_record) }
+  let(:person_record)  { instance_double(Person, type: person_type) }
+
+  let(:address_type) { nil }
+  let(:person_type)  { nil }
+
+  # NOTE: we only really have for now `Applicant`, but as these decorator is built
+  # to cope with multiple people types, we are testing it also for `Partner`
+
+  describe '#page_title' do
+    context 'for an applicant' do
+      let(:person_type) { Applicant.to_s }
+
+      context 'for a home address' do
+        let(:address_type) { HomeAddress.to_s }
+        it { expect(subject.page_title).to eq('.page_title.applicant.home_address') }
+      end
+
+      context 'for a correspondence address' do
+        let(:address_type) { CorrespondenceAddress.to_s }
+        it { expect(subject.page_title).to eq('.page_title.applicant.correspondence_address') }
+      end
+    end
+
+    context 'for a partner' do
+      let(:person_type) { Partner.to_s }
+
+      context 'for a home address' do
+        let(:address_type) { HomeAddress.to_s }
+        it { expect(subject.page_title).to eq('.page_title.partner.home_address') }
+      end
+
+      context 'for a correspondence address' do
```

Comment:
I don't expect this scenario ever to occur - we will have to check when we build the partner flow if we need to offer a partner correspondence address, but afaik we won't

Reply:
I know, but the code has to be tested to cope with any variant, any combination of applicant/partner/address. There is a comment at the beginning of these tests to make it clear:
```
# NOTE: we only really have for now `Applicant`, but as these decorator is built
# to cope with multiple people types, we are testing it also for `Partner`
```